lib/App/Rakubrew/Shell.pm

```diff
@@ -144,64 +151,94 @@ sub clean_path {
     return $path;
 }
 
+# Strips out all elements in arguments array up to and including $bre_name
+# command.  The first argument is index where the completion should look for the
+# word to be completed.
+sub strip_executable {
+    my $self = shift;
+    my $index = shift;
+
+    my $cmd_pos = 0;
+    foreach my $word (@_) {
+        ++$cmd_pos;
+        --$index;
+        last if $word =~ /(^|\W)$brew_name$/;
+    }
+    return ($index, @_[$cmd_pos..$#_])
+}
+
 =pod
 
 Returns a list of completion candidates.
 This function takes two parameters:
 
 =over 4
 
-=item * Index of the word to complete (starting at 0)
+=item * Index of the word to complete, 0-based. If C<-1> is passed then list of all commands is returned.
 
 =item * A list of words already entered
 
 =back
 
 =cut
+
+sub _filter_candidates {
+    my $self = shift;
+    my $seed = shift;
+    return 
+        # If a shell preserves ordering then put the prefix-mathing candidates first. I.e. for 'ver' 'version' would
+        # precede 'rakudo-version'
+        sort { index($a, $seed) cmp index($a, $seed) }
```

Comment:
Shouldn't the second `$a` be `$b` instead?

Reply:
Indeed. Fixed, thanks!